### PR TITLE
Documentation for Binary Quantization Support with KNN Vector Search

### PR DESCRIPTION
**Pull request by Vikasht34**
### Description
This PR introduces detailed documentation for Binary Quantization support in the k-NN plugin, including how to configure, fine-tune, and optimize k-NN vector indices using Binary Quantization. Key highlights of this update are as follows:
**Key Changes:**
1. Introduction to Binary Quantization:

    - Binary Quantization allows vectors to be quantized into binary representations (1-bit, 2-bit, or 4-bit) to reduce memory footprint and speed up search performance.
    - It provides compression rates of 32x, 16x, and 8x respectively for 1-bit, 2-bit, and 4-bit quantization, drastically reducing memory usage.
2. Out-of-the-Box Experience:
    - Default 1-bit binary quantization is supported without any additional configuration, offering 32x compression and default parameters for ef_search and ef_construction (both set to 100).
3. Customization and Fine-tuning: Users can further optimize the indexing and search processes by specifying custom parameters like:
      - ef_construction: Adjusting the construction of HNSW graphs.
      - compression_level: Providing different compression levels such as 32x for 1-bit, 16x for 2-bit, and 8x for 4-bit.
      -  oversample_factor: Fine-tuning search accuracy by oversampling neighbors during the search process.    
          
4. Memory Estimation for Binary Quantization:
    - Provided memory estimation formulas for HNSW graph construction for 1-bit, 2-bit, and 4-bit quantization, helping users estimate the memory requirements based on their vector dimensions and index configurations.

5. Example Configurations:
      - Basic Binary Quantization Example: Demonstrates how to configure Binary Quantization with minimal setup and default 1-bit quantization.
      - Advanced Configurations: Shows how to customize parameters like compression level and ef_construction to achieve optimal performance for different use cases.

### Issues Resolved
Closes #[_insert issue number_]

### Version
2.17

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _search-plugins/knn/knn-vector-quantization.md [hunk 1]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+## Binary Quantization
+
+Starting with the version 2.17, OpenSearch supports Binary Quantization (BQ) with binary vector support from the Faiss engine. Binary quantization compresses vectors into a binary format (0s and 1s), making it highly efficient in terms of memory usage. You can choose to represent each vector dimension using 1, 2, or 4 bits, depending on the precision you want. One of the advantages of using Binary Quantization (BQ) is that the training process is handled automatically during indexing. This means that no separate training step is required, unlike other quantization techniques such as Product Quantization (PQ).
+
+### Using Binary quantization
+To use Binary Quantization in your k-NN vector index, you can configure it with minimal effort. Below is an example of how you can define a k-NN vector field that utilizes Binary Quantization with the Faiss engine. This configuration provides an out-of-the-box setup with 1-bit binary quantization, using the default values for ef_search and ef_construction set to 100:
+```json
+PUT my-vector-index
+{
+  "mappings": {
```
Comment (jmazanec15) [inline _search-plugins/knn/knn-vector-quantization.md:322]: I dont think we need to include these on-disk/compression examples - just use the low level interface

**Comment by kolchfa-aws** [comment]
@Vikasht34 Please let me know when this PR is ready for doc review. Thanks!

File: _search-plugins/knn/knn-vector-quantization.md (hunk 1 above)
Comment (Vikasht34) [reply _search-plugins/knn/knn-vector-quantization.md:322]: IMO , I was thinking as user , When I go to Quantization Page , I know how to configure and what defaults are there , so I feel we should keep to have all information at one place.

**Comment by Vikasht34** [comment]
> @Vikasht34 Please let me know when this PR is ready for doc review. Thanks!

Please start review!!

File: _search-plugins/knn/knn-vector-quantization.md [hunk 2]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+                }
+              }
+            }
+        }
+      }
+    }
+  }
+}
+```
+### Basic Search with k-NN and Binary Quantization
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:395]:
```suggestion
### Search using binary quantized vectors
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 3]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+You can perform a basic k-NN search on your index using a vector and specifying the number of nearest neighbors (k) to return:
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:396]:
```suggestion
You can perform a k-NN search on your index by providing a vector and specifying the number of nearest neighbors (k) to return:

```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 4]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+  "query": {
+    "knn": {
+      "my_vector_field": {
+        "vector": [1.5, 5.5, 1.5, 5.5, 1.5, 5.5, 1.5, 5.5],
+        "k": 10
+      }
+    }
+  }
+}
+```
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:410]:
````suggestion
```
{% include copy-curl.html %}

````

File: _search-plugins/knn/knn-vector-quantization.md [hunk 5]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+You can also fine-tune the search process by adjusting the ef_search and oversample_factor parameters. Here's an example of how to do this:
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:411]:
```suggestion
You can also fine-tune search by providing the `ef_search` and `oversample_factor` parameters
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 6]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+- **`oversample_factor`**: This parameter controls the factor by which the search process oversamples the candidate vectors before ranking them. A higher oversample factor means more candidates will be considered before ranking, improving accuracy but also increasing search time. Essentially, it helps trade-off between accuracy and efficiency. For example, setting the `oversample_factor` to `2.0` will double the number of candidates considered during the ranking phase, which may help achieve better results.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:412]:
```suggestion
The `oversample_factor` parameter controls the factor by which the search oversamples the candidate vectors before ranking them. A higher oversample factor means more candidates will be considered before ranking, improving accuracy but also increasing search time. When selecting the `oversample_factor` value, consider the trade-off between accuracy and efficiency. For example, setting the `oversample_factor` to `2.0` will double the number of candidates considered during the ranking phase, which may help achieve better results. 

The following request specifies the `ef_search` and `oversample_factor` parameters:

```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 7]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+            "ef_search": 10
+        },
+        "rescore": {
+            "oversample_factor": 10.0
+        }
+      }
+    }
+  }
+}
+
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:432]:
```suggestion
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 8]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+```
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:433]:
````suggestion
```
{% include copy-curl.html %}

````

File: _search-plugins/knn/knn-vector-quantization.md [hunk 9]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+
+#### HNSW memory estimation
+
+The memory required for the Hierarchical Navigable Small World (HNSW) graph can be estimated as `1.1 * (dimension + 8 * M)` bytes/vector, where `M` is the maximum number of bidirectional links created for each element during the construction of the graph.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:437]:
```suggestion
The memory required for the Hierarchical Navigable Small World (HNSW) graph can be estimated as `1.1 * (dimension + 8 * m)` bytes/vector, where `m` is the maximum number of bidirectional links created for each element during the construction of the graph.
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 10]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+
+As an example, assume that you have 1 million vectors with a dimension of 256 and M of 16. The memory requirement can be estimated as follows:
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:439]:
```suggestion
As an example, assume that you have 1 million vectors with a dimension of 256 and an `m` of 16. The following sections provide memory requirement estimations for various compression values.
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 11]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+
+##### 1-bit Quantization (32x Compression)
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:441]:
```suggestion
##### 1-bit quantization (32x compression)
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 12]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+In this case, each dimension is represented using 1 bit, equivalent to a 32x compression factor.
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:442]:
```suggestion
In 1-bit quantization, each dimension is represented using 1 bit, equivalent to a 32x compression factor. The memory requirement can be estimated as follows:
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 13]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+
+```r
+Memory = 1.1 * ((256 * 1 / 8) + 8 * 16) * 1,000,000
+       ~= 0.176 GB
+```
+##### 2-bit Quantization (16x Compression)
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:448]:
```suggestion

##### 2-bit quantization (16x compression)
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 14]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+```r
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:449]:
````suggestion
In 2-bit quantization, each dimension is represented using 2 bits, equivalent to a 16x compression factor. The memory requirement can be estimated as follows:

```r
````

File: _search-plugins/knn/knn-vector-quantization.md [hunk 15]
```diff
@@ -310,3 +310,148 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+Memory = 1.1 * ((256 * 2 / 8) + 8 * 16) * 1,000,000
+       ~= 0.211 GB
+```
+##### 4-bit Quantization (8x Compression)
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:453]:
```suggestion

##### 4-bit quantization (8x compression)

In 4-bit quantization, each dimension is represented using 4 bits, equivalent to a 8x compression factor. The memory requirement can be estimated as follows:

```

**Review by kolchfa-aws — APPROVED** [review]
Thanks, @Vikasht34! I made several rewrites. In the future, please work on a branch in your fork (not on main) because then maintainers can push the rewrites directly into your PR.

File: _search-plugins/knn/knn-vector-quantization.md [hunk 16]
```diff
@@ -310,3 +310,170 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+PUT my-vector-index
+{
+  "mappings": {
+    "properties": {
+      "my_vector_field": {
+        "type": "knn_vector",
+        "dimension": 8,
+        "space_type": "l2",
+        "data_type": "float",
+        "mode": "on-disk",
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:358]:
```suggestion
        "mode": "on_disk",
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 17]
```diff
@@ -310,3 +310,170 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+{% include copy-curl.html %}
+
+
+#### HNSW memory estimation
+
+The memory required for the Hierarchical Navigable Small World (HNSW) graph can be estimated as `1.1 * (dimension + 8 * m)` bytes/vector, where `m` is the maximum number of bidirectional links created for each element during the construction of the graph.
+
+As an example, assume that you have 1 million vectors with a dimension of 256 and an `m` of 16. The following sections provide memory requirement estimations for various compression values.
+
+##### 1-bit quantization (32x compression)
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:456]:
```suggestion
##### 1-bit quantization (32x compression)

```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 18]
```diff
@@ -310,3 +310,170 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+In 1-bit quantization, each dimension is represented using 1 bit, equivalent to a 32x compression factor. The memory requirement can be estimated as follows:
+
+```r
+Memory = 1.1 * ((256 * 1 / 8) + 8 * 16) * 1,000,000
+       ~= 0.176 GB
+```
+
+##### 2-bit quantization (16x compression)
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:464]:
```suggestion
##### 2-bit quantization (16x compression)

```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 19]
```diff
@@ -310,3 +310,172 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+                "ef_construction": 16
+            }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+The following example futher fine-tunes the configuration by defining `ef_construction` , `encoder` and the number of bits `bits`:
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:371]:
```suggestion

The following example futher fine-tunes the configuration by defining `ef_construction` , `encoder` and the number of bits `bits`:
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 20]
```diff
@@ -310,3 +310,173 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
 ```r
 1.1*((8 / 8 * 64 + 24) * 1000000  + 100 * (2^8 * 4 * 256 + 4 * 512 * 256))  ~= 0.171 GB
 ```
+
+## Binary Quantization
+
+Starting with the version 2.17, OpenSearch supports binary quantization (BQ) with binary vector support for the Faiss engine. Binary quantization compresses vectors into a binary format (0s and 1s), making it highly efficient in terms of memory usage. You can choose to represent each vector dimension using 1, 2, or 4 bits, depending on the desired precision. One of the advantages of using binary quantization is that the training process is handled automatically during indexing. This means that no separate training step is required, unlike other quantization techniques such as product quantization.
+
+### Using binary quantization
+To configure binary quantization for the Faiss engine, define a `knn_vector` field and specify the `mode` as `on_disk`. This configuration defaults to 1-bit binary quantization and both `ef_search` and `ef_construction` set to `100`:
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:319]:
```suggestion
To configure binary quantization for the Faiss engine, define a `knn_vector` field and specify the `mode` as `on_disk`. This configuration defaults to 1-bit binary quantization and both `ef_search` and `ef_construction` set to `100`:

```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 21]
```diff
@@ -310,3 +310,174 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+            }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The following example futher fine-tunes the configuration by defining `ef_construction` , `encoder` and the number of bits `bits`:
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:373]:
```suggestion
The following example futher fine-tunes the configuration by defining `ef_construction` , `encoder` and the number of bits `bits` (which can be `1`, `2`, or `4`):
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 22]
```diff
@@ -310,3 +310,174 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+            "name": "hnsw",
+            "engine": "faiss",
+            "space_type": "l2",
+            "params": {
+              "m": 16,
+              "ef_construction": 512,
+              "encoder": {
+                "name": "binary",
+                "parameters": {
+                  "bits": 1  // Can be 1, 2, or 4
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:393]:
```suggestion
                  "bits": 1 
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 23]
```diff
@@ -310,3 +310,174 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+            }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Search using binary quantized vectors
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:405]:
```suggestion
### Search using binary quantized vectors

```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 24]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
 ```r
 1.1*((8 / 8 * 64 + 24) * 1000000  + 100 * (2^8 * 4 * 256 + 4 * 512 * 256))  ~= 0.171 GB
 ```
+
+## Binary Quantization
+
+Starting with the version 2.17, OpenSearch supports binary quantization (BQ) with binary vector support for the Faiss engine. Binary quantization compresses vectors into a binary format (0s and 1s), making it highly efficient in terms of memory usage. You can choose to represent each vector dimension using 1, 2, or 4 bits, depending on the desired precision. One of the advantages of using binary quantization is that the training process is handled automatically during indexing. This means that no separate training step is required, unlike other quantization techniques such as product quantization.
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:316]:
```suggestion
Starting with version 2.17, OpenSearch supports BQ with binary vector support for the Faiss engine. BQ compresses vectors into a binary format (0s and 1s), making it highly efficient in terms of memory usage. You can choose to represent each vector dimension using 1, 2, or 4 bits, depending on the desired precision. One of the advantages of using BQ is that the training process is handled automatically during indexing. This means that no separate training step is required, unlike other quantization techniques such as PQ.
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 25]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+
+### Using binary quantization
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:318]:
```suggestion
### Using BQ
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 26]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+To configure binary quantization for the Faiss engine, define a `knn_vector` field and specify the `mode` as `on_disk`. This configuration defaults to 1-bit binary quantization and both `ef_search` and `ef_construction` set to `100`:
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:319]:
```suggestion
To configure BQ for the Faiss engine, define a `knn_vector` field and specify the `mode` as `on_disk`. This configuration defaults to 1-bit BQ and both `ef_search` and `ef_construction` set to `100`:
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 27]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+        "data_type": "float",
+        "mode": "on_disk"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+To further optimize the configuration, you can specify additional parameters such as the compression level and fine-tune the search parameters. For example, you can override the `ef_construction` value or define the compression level, which corresponds to the number of bits used for quantization:
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:339]:
```suggestion
To further optimize the configuration, you can specify additional parameters, such as the compression level, and fine-tune the search parameters. For example, you can override the `ef_construction` value or define the compression level, which corresponds to the number of bits used for quantization:
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 28]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+            }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The following example futher fine-tunes the configuration by defining `ef_construction` , `encoder` and the number of bits `bits` (which can be `1`, `2`, or `4`):
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:373]:
```suggestion
The following example further fine-tunes the configuration by defining `ef_construction`, `encoder`, and the number of `bits` (which can be `1`, `2`, or `4`):
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 29]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+        "vector": [1.5, 5.5, 1.5, 5.5, 1.5, 5.5, 1.5, 5.5],
+        "k": 10
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+You can also fine-tune search by providing the `ef_search` and `oversample_factor` parameters
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:425]:
```suggestion
You can also fine-tune search by providing the `ef_search` and `oversample_factor` parameters.
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 30]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+The `oversample_factor` parameter controls the factor by which the search oversamples the candidate vectors before ranking them. A higher oversample factor means more candidates will be considered before ranking, improving accuracy but also increasing search time. When selecting the `oversample_factor` value, consider the trade-off between accuracy and efficiency. For example, setting the `oversample_factor` to `2.0` will double the number of candidates considered during the ranking phase, which may help achieve better results.
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:426]:
```suggestion
The `oversample_factor` parameter controls the factor by which the search oversamples the candidate vectors before ranking them. Using a higher oversample factor means that more candidates will be considered before ranking, improving accuracy but also increasing search time. When selecting the `oversample_factor` value, consider the trade-off between accuracy and efficiency. For example, setting the `oversample_factor` to `2.0` will double the number of candidates considered during the ranking phase, which may help achieve better results. 
```

File: _search-plugins/knn/knn-vector-quantization.md [hunk 31]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
...
+In 2-bit quantization, each dimension is represented using 2 bits, equivalent to a 16x compression factor. The memory requirement can be estimated as follows:
+
+```r
+Memory = 1.1 * ((256 * 2 / 8) + 8 * 16) * 1,000,000
+       ~= 0.211 GB
+```
+
+##### 4-bit quantization (8x compression)
+
+In 4-bit quantization, each dimension is represented using 4 bits, equivalent to a 8x compression factor. The memory requirement can be estimated as follows:
```
Comment (natebower) [inline _search-plugins/knn/knn-vector-quantization.md:479]:
```suggestion
In 4-bit quantization, each dimension is represented using 4 bits, equivalent to an 8x compression factor. The memory requirement can be estimated as follows:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @Vikasht34 Please see my changes and let me know if you have any questions. Thanks!

File: _search-plugins/knn/knn-vector-quantization.md [hunk 32]
```diff
@@ -310,3 +310,175 @@ For example, assume that you have 1 million vectors with a dimension of 256, `iv
 ```r
 1.1*((8 / 8 * 64 + 24) * 1000000  + 100 * (2^8 * 4 * 256 + 4 * 512 * 256))  ~= 0.171 GB
 ```
+
+## Binary Quantization
```
Comment (kolchfa-aws) [inline _search-plugins/knn/knn-vector-quantization.md:314]:
```suggestion
## Binary quantization
```

**Pull request merged**